cpp_api/options.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+class optionst;
+
+class api_optionst
```

Comment:
At what stage does validation of the supplied options happen? The singular bool option defined now doesn't really need validating. But I foresee a need to report errors if options are specified which don't make sense in combination, or to report warnings if a deprecated option is used.